src/classes/i18n.js

```diff
@@ -199,15 +209,94 @@ export default class i18n {
     }
 
     get_lang_files (lang, cb) {
+        const MAX_RETRY = this.config.get('download_retry');
 
         // append a random string, to avoid getting a response from cache
         const random_str = '&rand=' + ~~(Math.random() * 1000);
         const url = this.translations_url.replace(':lang', lang) + random_str;
-        const write_stream = fs.createWriteStream(this.config.get('locale_dir') + lang + '.json');
 
-        request(url)
-            .pipe(write_stream)
-            .on('finish', cb);
+        const translation_file_path = this.config.get('locale_dir') + lang + '.json';
+
+        new Promise(async (resolve, reject) => {
```

Comment:
I thought translation server has ability to serve a zipped package of all translation?